### ISSUE TITLE
[TF2.0] Load model with tf.keras.models.load_model does not work

### ISSUE THREAD

**Issue by quetil** [tensorflow/tensorflow#28281]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No comes from stock example script https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/generative/pix2pix.ipynb
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10/RHEL 7
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: --
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.12.0-9492-g2c319fb415 2.0.0-alpha0
- **Python version**: 3.7.3
- **Bazel version (if compiling from source)**: --
- **GCC/Compiler version (if compiling from source)**: --
- **CUDA/cuDNN version**: both, without and with CUDA 10/cuDNN 7.5
- **GPU model and memory**: both, without and with P40 / 8192MiB
### Exact command to reproduce
```python
import tensorflow as tf


OUTPUT_CHANNELS = 3

def downsample(filters, size, apply_batchnorm=True):
  initializer = tf.random_normal_initializer(0., 0.02)

  result = tf.keras.Sequential()
  result.add(
      tf.keras.layers.Conv2D(filters, size, strides=2, padding='same',
                             kernel_initializer=initializer, use_bias=False))

  if apply_batchnorm:
    result.add(tf.keras.layers.BatchNormalization())

  result.add(tf.keras.layers.LeakyReLU())

  return result

def upsample(filters, size, apply_dropout=False):
  initializer = tf.random_normal_initializer(0., 0.02)

  result = tf.keras.Sequential()
  result.add(
    tf.keras.layers.Conv2DTranspose(filters, size, strides=2,
                                    padding='same',
                                    kernel_initializer=initializer,
                                    use_bias=False))

  result.add(tf.keras.layers.BatchNormalization())

  if apply_dropout:
      result.add(tf.keras.layers.Dropout(0.5))

  result.add(tf.keras.layers.ReLU())

  return result


def Generator():
  down_stack = [
    downsample(64, 4, apply_batchnorm=False), # (bs, 128, 128, 64)
    downsample(128, 4), # (bs, 64, 64, 128)
    downsample(256, 4), # (bs, 32, 32, 256)
    downsample(512, 4), # (bs, 16, 16, 512)
    downsample(512, 4), # (bs, 8, 8, 512)
    downsample(512, 4), # (bs, 4, 4, 512)
    downsample(512, 4), # (bs, 2, 2, 512)
    downsample(512, 4), # (bs, 1, 1, 512)
  ]

  up_stack = [
    upsample(512, 4, apply_dropout=True), # (bs, 2, 2, 1024)
    upsample(512, 4, apply_dropout=True), # (bs, 4, 4, 1024)
    upsample(512, 4, apply_dropout=True), # (bs, 8, 8, 1024)
    upsample(512, 4), # (bs, 16, 16, 1024)
    upsample(256, 4), # (bs, 32, 32, 512)
    upsample(128, 4), # (bs, 64, 64, 256)
    upsample(64, 4), # (bs, 128, 128, 128)
  ]

  initializer = tf.random_normal_initializer(0., 0.02)
  last = tf.keras.layers.Conv2DTranspose(OUTPUT_CHANNELS, 4,
                                         strides=2,
                                         padding='same',
                                         kernel_initializer=initializer,
                                         activation='tanh') # (bs, 256, 256, 3)

  concat = tf.keras.layers.Concatenate()

  inputs = tf.keras.layers.Input(shape=[None,None,3])
  x = inputs

  # Downsampling through the model
  skips = []
  for down in down_stack:
    x = down(x)
    skips.append(x)

  skips = reversed(skips[:-1])

  # Upsampling and establishing the skip connections
  for up, skip in zip(up_stack, skips):
    x = up(x)
    x = concat([x, skip])

  x = last(x)

  return tf.keras.Model(inputs=inputs, outputs=x)

generator = Generator()
generator.summary()

generator.save('generator.h5')
generator_loaded = tf.keras.models.load_model('generator.h5')
```

### Describe the problem
The saving works apparently well and generates a file. But, impossible to load the model (no error message, the command never ends).

### Source code / logs
Codes come from pix2pix example (https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/generative/pix2pix.ipynb)

**Comment by quetil** [comment]
Update: 
I think the issue occurs with the sequential API. By using the funcitonal API, ```tf.keras.models.load_model``` works well. It's a temporary solution until a fix will be released.
If anyone can confirm my assumption
Cheers,
AQ

**Comment by rpmcruz** [comment]
Here is a more concise reproduction of the bug:

```
import tensorflow as tf

seq = tf.keras.Sequential()
seq.add(tf.keras.layers.Dense(3))  # , input_shape=(5,)

x = input_layer = tf.keras.layers.Input((5,))
x = seq(x)
x = tf.keras.layers.Dense(3)(x)
model = tf.keras.Model(input_layer, x)

print('Saving...')
model.save('model.h5')
print('Loading...')
tf.keras.models.load_model('model.h5')
print('Done')
```

The problem happens when using a `Sequential` without an `input_shape`. TensorFlow allows such models to be used inside other models. Unfortunately, `load_model()` has not been implemented to work in such a scenario. (It seems to enter into an infinite loop.)

**Comment by rpmcruz** [comment]
I don't know how to add labels, but this is still a problem in `2.0.0-beta1`.

**Comment by wwwxxx2016** [comment]
I have the same problem, when I set the 'input_shape' then the load_model is OK.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly ''2.0.0-dev20190723'. Thanks!
